<a href="https://colab.research.google.com/github/shannon3851/workstuff/blob/main/Res2_potentialdensityanalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **RES2 - POTENTIAL DENSITY 8DU/ACRE**

In [10]:
import pandas as pd

file_path = '/content/RES2_version3_datatable.xls'
res2 = pd.read_excel(file_path)
print(res2.head(2))
print(len(res2))

  Owner Comment Link PARCELID UNIQUE_ID Subdivisio  LocationPr ParishID  \
0                       17803     17803                   Lilo        6   
1                       20876     20876             Palm Creek        5   

                                  Property_N  VacantLot  ...     YCOORD  \
0   Lilo, 2 Long Point Road, Cavello Bay, Sy          0  ...  133088.59   
1  Palm Creek, 6 Grasmere Rd, Fairylands, Pm          0  ...  133324.00   

                  Full_Addre Units    Predominat2     Tax_Code    Asst__num  \
0  2 Long Point Lane, Sandys   2.0  Residential 2  Residential   31473016.0   
1  6 Grasmere Road, Pembroke   2.0  Residential 2  Residential  121392015.0   

               Building_N  Property_D  TRA__sm_  UnitNumber  
0                 LI - LO       HOUSE     244.0         1.0  
1  Palm Greek - Upper Apt   Apartment     217.0         1.0  

[2 rows x 34 columns]
4100


In [11]:
print(res2['Density'].describe())

count    4100.000000
mean        4.924898
std        10.450007
min         0.000000
25%         1.597042
50%         3.111114
75%         5.731011
max       503.790169
Name: Density, dtype: float64


In [12]:
# number parcels below 8du/acre
below = (res2['Density']<=8).sum()
print('below', below)

# % parcels below 8du/acre
below_percent = ((below/4100)*100)
print('percent below:', below_percent)

below 3485
percent below: 85.0


In [13]:
# parcels above 8du/acre
above = (res2['Density']>=8).sum()
print('above', above)

#percent above 8du/acre
above_percent = ((above/4100)*100)
print('percent above:', above_percent)

above 615
percent above: 15.0


In [14]:
# acres below 8du/acre
acres_below = res2.loc[res2['Density']<=8, 'Acreage'].sum()
print('acres below:', acres_below)

#acres above 8du/acre
acres_above = res2.loc[res2['Density']>=8, 'Acreage'].sum()
print('acres above:', acres_above)

acres below: 2471.4575558127835
acres above: 177.01456969603896


In [15]:
# units below 8du/acre
units_below = res2.loc[res2['Density']<=8, 'Join_Count'].sum()
print('units below:', units_below)


# units above 8du/acre
units_above = res2.loc[res2['Density']>=8, 'Join_Count'].sum()
print('units above:', units_above)

units below: 5571
units above: 2541


In [16]:
print(res2['Join_Count'].sum())

8112


In [17]:
# export parcels below 8du/acre
acres_below_export = res2.loc[res2['Density']<=8]

print(acres_below_export)

acres_below_export.to_excel('R2_below8du.xlsx', index=False)




     Owner                                            Comment  \
0                                                               
1                                                               
2                                                               
3                                                               
4                                                               
...    ...                                                ...   
4094                                                            
4096        Split from 28954 after review of Zoning Order ...   
4097                                                            
4098                                                            
4099                                                            

                                                   Link PARCELID UNIQUE_ID  \
0                                                          17803     17803   
1                                                          2087

In [18]:
# export parcels above 8du/acre
acres_above_export = res2.loc[res2['Density']>=8]
print(acres_below_export)

acres_above_export.to_excel('R2_above8du.xlsx', index=False)

     Owner                                            Comment  \
0                                                               
1                                                               
2                                                               
3                                                               
4                                                               
...    ...                                                ...   
4094                                                            
4096        Split from 28954 after review of Zoning Order ...   
4097                                                            
4098                                                            
4099                                                            

                                                   Link PARCELID UNIQUE_ID  \
0                                                          17803     17803   
1                                                          2087

PARCEL LAYER HAS 3085 ROWS....EXCEL XY TABLE HAS 3485....WHERE ARE THE 400 INPUTS GOING // WHY AREN'T THEY BEING TRANSFERED OVER TO GIS LAYER PROPERLY???

In [20]:
import pandas as pd

#PARCEL LAYER

file_path = '/content/GISlayer_res2_tablejoin_below8du.xls'
below8du = pd.read_excel(file_path)
print(below8du.head(1))

print(len(below8du))

   Parcels_Sept2025.FID  Parcels_Sept2025.OBJECTID Parcels_Sept2025.Owner  \
0                    88                         89                          

  Parcels_Sept2025.Comment Parcels_Sept2025.Link  Parcels_Sept2025.PARCELID  \
0                                                                     17803   

   Parcels_Sept2025.UNIQUE_ID Parcels_Sept2025.Subdivisio  \
0                       17803                               

  Parcels_Sept2025.LocationPr Parcels_Sept2025.ParishID  ...  \
0                        Lilo                         6  ...   

          Sheet1$.Full_Addre  Sheet1$.Units  Sheet1$.Predominat2  \
0  2 Long Point Lane, Sandys            2.0        Residential 2   

   Sheet1$.Tax_Code Sheet1$.Asst__num Sheet1$.Building_N Sheet1$.Property_D  \
0       Residential        31473016.0            LI - LO              HOUSE   

   Sheet1$.TRA__sm_  Sheet1$.UnitNumber  Sheet1$.ObjectID  
0             244.0                 1.0                 1  

[1 rows x 74 colu

In [21]:
# XY TABLE

file_path = '/content/below8du_table_xypoint.xls'
below_xy = pd.read_excel(file_path)
print(below_xy.head(1))

print(len(below_xy))

   OBJECTID Owner Comment Link PARCELID UNIQUE_ID Subdivisio LocationPr  \
0         1                       17803     17803                  Lilo   

  ParishID                                Property_N  ...     YCOORD  \
0        6  Lilo, 2 Long Point Road, Cavello Bay, Sy  ...  133088.59   

                  Full_Addre  Units    Predominat2     Tax_Code   Asst__num  \
0  2 Long Point Lane, Sandys    2.0  Residential 2  Residential  31473016.0   

   Building_N  Property_D  TRA__sm_  UnitNumber  
0     LI - LO       HOUSE     244.0         1.0  

[1 rows x 33 columns]
3485


In [22]:
print(below8du.describe())

       Parcels_Sept2025.FID  Parcels_Sept2025.OBJECTID  \
count           3085.000000                3085.000000   
mean            8385.476823                8393.691086   
std             5241.750272                5247.413476   
min               88.000000                  89.000000   
25%             3521.000000                3526.000000   
50%             8004.000000                8009.000000   
75%            12532.000000               12542.000000   
max            20957.000000               21038.000000   

       Parcels_Sept2025.PARCELID  Parcels_Sept2025.UNIQUE_ID  \
count                3085.000000                 3085.000000   
mean                28654.265802                28654.265802   
std                  8822.203829                 8822.203829   
min                  3016.000000                 3016.000000   
25%                 21218.000000                21218.000000   
50%                 28974.000000                28974.000000   
75%                 36366.000

In [23]:
below_xy['UNIQUE_ID'].isin(below8du['Parcels_Sept2025.UNIQUE_ID']).value_counts()

,count
UNIQUE_ID,
False,3485


In [24]:
below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID']).value_counts()

,count
Parcels_Sept2025.UNIQUE_ID,
False,3085


In [25]:
print(
    below8du.loc[
        below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID']),
        'Parcels_Sept2025.UNIQUE_ID'
    ]
)

Series([], Name: Parcels_Sept2025.UNIQUE_ID, dtype: int64)


In [26]:
print(below8du['Parcels_Sept2025.UNIQUE_ID'].dtype)
print(below_xy['UNIQUE_ID'].dtype)

int64
object


In [27]:
not_in_layer = below_xy[below_xy['UNIQUE_ID'].isin(below8du['Parcels_Sept2025.UNIQUE_ID'])]
print(not_in_layer)

Empty DataFrame
Columns: [OBJECTID, Owner, Comment, Link, PARCELID, UNIQUE_ID, Subdivisio, LocationPr, ParishID, Property_N, VacantLot, Acreage, Area_SqFt, Predominat, Zoning_Ord, Developmen, Join_Count, Density, TARGET_FID, ADDRESSID, CIVICNUMBE, POSTALCODE, XCOORD, YCOORD, Full_Addre, Units, Predominat2, Tax_Code, Asst__num, Building_N, Property_D, TRA__sm_, UnitNumber]
Index: []

[0 rows x 33 columns]


In [28]:
below_xy[below_xy['UNIQUE_ID'].str.strip().str.isnumeric()]

,OBJECTID,Owner,Comment,Link,PARCELID,UNIQUE_ID,Subdivisio,LocationPr,ParishID,Property_N,...,YCOORD,Full_Addre,Units,Predominat2,Tax_Code,Asst__num,Building_N,Property_D,TRA__sm_,UnitNumber
0,1,,,,17803,17803,,Lilo,6,"Lilo, 2 Long Point Road, Cavello Bay, Sy",...,133088.59,"2 Long Point Lane, Sandys",2.0,Residential 2,Residential,31473016.0,LI - LO,HOUSE,244.0,1.0
1,2,,,,20876,20876,,Palm Creek,5,"Palm Creek, 6 Grasmere Rd, Fairylands, Pm",...,133324.00,"6 Grasmere Road, Pembroke",2.0,Residential 2,Residential,121392015.0,Palm Greek - Upper Apt,Apartment,217.0,1.0
2,3,,,,21768,21768,,Windward,2,,...,132905.00,"21 Tee Street, Devonshire",2.0,Residential 2,Residential,41845013.0,Al Castello,House,527.0,1.0
3,4,,,,40148,40148,,Idolwodd Lagoon Cottage,9,,...,137174.51,"2 Shell Point Road, St. George's",1.0,Residential 2,Residential,50378015.0,IDOLWOOD LAGOON COTTAGE,HOUSE,162.0,1.0
4,5,,,,33714,33714,,Broadreach,5,,...,133551.25,"10 Hallett Crescent, Pembroke",2.0,Residential 2,Residential,121407012.0,MAIN UNIT - BROADREACH,HOUSE,247.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3480,3481,,,,44804,44804,,,5,,...,133545.57,"32 Shaw Wood Road, Pembroke",1.0,Residential 2,Residential,125138512.0,Ballast Cottage,House,246.0,1.0
3481,3482,,Split from 28954 after review of Zoning Order ...,http://www.royalgazette.com/article/20130703/N...,44807,44807,,Gatewood,4,,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3482,3483,,,,44809,44809,S3387,,11,,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3483,3484,,,,44810,44810,,,4,,...,131428.26,"3 Simons Lane, Paget",1.0,Residential 2,Tourist,71582010.0,Valley Cottages - Tourist,Tourist Apartment,452.0,1.0


In [29]:
print(below_xy['UNIQUE_ID'].isna().sum())

0


In [30]:
print(below8du['Parcels_Sept2025.UNIQUE_ID'].dtype)
print(below_xy['UNIQUE_ID'].dtype)

int64
object


In [31]:
not_in_xy = below8du[below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID'])]
print(not_in_xy)

Empty DataFrame
Columns: [Parcels_Sept2025.FID, Parcels_Sept2025.OBJECTID, Parcels_Sept2025.Owner, Parcels_Sept2025.Comment, Parcels_Sept2025.Link, Parcels_Sept2025.PARCELID, Parcels_Sept2025.UNIQUE_ID, Parcels_Sept2025.Subdivisio, Parcels_Sept2025.LocationPr, Parcels_Sept2025.ParishID, Parcels_Sept2025.Property_N, Parcels_Sept2025.VacantLot, Parcels_Sept2025.Acreage, Parcels_Sept2025.Area_SqFt, Parcels_Sept2025.Predominat, Parcels_Sept2025.Zoning_Ord, Parcels_Sept2025.Developmen, Parcels_Sept2025.Acres, Parcels_Sept2025.Shape_Leng, Parcels_Sept2025.Shape_Area, V3_residential2_wVacant_AddSpatialJoin.OBJECTID, V3_residential2_wVacant_AddSpatialJoin.Join_Count, V3_residential2_wVacant_AddSpatialJoin.TARGET_FID, V3_residential2_wVacant_AddSpatialJoin.ADDRESSID, V3_residential2_wVacant_AddSpatialJoin.CIVICNUMBE, V3_residential2_wVacant_AddSpatialJoin.POSTALCODE, V3_residential2_wVacant_AddSpatialJoin.XCOORD, V3_residential2_wVacant_AddSpatialJoin.YCOORD, V3_residential2_wVacant_AddSpatialJ

In [32]:
count = (below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID'])).sum()
print(count)

0


In [33]:
print(
below8du.loc[
below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID']),
'Parcels_Sept2025.UNIQUE_ID'
]
)

Series([], Name: Parcels_Sept2025.UNIQUE_ID, dtype: int64)


In [34]:
missing_mask = below_xy['UNIQUE_ID'].isin(below8du['Parcels_Sept2025.UNIQUE_ID'])

print(missing_mask.sum())

0


In [35]:
missing_mask = below8du['Parcels_Sept2025.UNIQUE_ID'].isin(below_xy['UNIQUE_ID'])

print(missing_mask.sum())

0


In [36]:
len(below8du)
len(below_xy)

below8du['Parcels_Sept2025.UNIQUE_ID'].nunique()
below_xy['UNIQUE_ID'].nunique()

below_xy['UNIQUE_ID'].isin(below8du['Parcels_Sept2025.UNIQUE_ID']).value_counts()

,count
UNIQUE_ID,
False,3485


In [39]:
print("Rows in below_xy:", len(below_xy))
print("Matched:", below_xy['UNIQUE_ID'].isin(
    below8du['Parcels_Sept2025.UNIQUE_ID']
).sum())
print("Missing:", (below_xy['UNIQUE_ID'].isin(
    below8du['Parcels_Sept2025.UNIQUE_ID']
)).sum())

Rows in below_xy: 3485
Matched: 0
Missing: 0


In [40]:
du8below = (below_xy['Density']<=8).sum()
print(du8below)

3485


In [41]:
layer_below8du = (below8du['Sheet1$.Density']<=8).sum()
print(layer_below8du)

3085


In [42]:
print("Rows in below8du:", len(below8du))
print("Matched:", below8du['Parcels_Sept2025.UNIQUE_ID'].isin(
    below_xy['UNIQUE_ID']
).sum())
print("Missing:", (below8du['Parcels_Sept2025.UNIQUE_ID'].isin(
    below_xy['UNIQUE_ID']
)).sum())

Rows in below8du: 3085
Matched: 0
Missing: 0
